### PR TITLE
Cleanup register endpoint

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

eiffel-community/etos#83

### Description of the Change

This PR is the first of several clean-up changes that will be created. I do them in steps to make reviewing easier.

This is done in preparation for the external provider change that is on its way ( #38 )
By making the endpoints cleaner, we can make the external provider changes easier.

### Alternate Designs

There are multiple designs, I chose the one where we

1. Reduce the size of webserver.py the most.
2. Make it easier to test.

### Benefits

Makes it easier to understand what is going on in the webserver.py file and makes it a lot easier to test.

### Possible Drawbacks

Some code-jumping will be involved. With proper function naming this will be mitigated.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: tests/test_webserver_register.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        fake_database = FakeDatabase()
+        fake_request = FakeRequest()
+        fake_response = FakeResponse()
+        self.logger.info("STEP: Send a register request with no providers.")
+        with self.assertRaises(falcon.HTTPBadRequest):
+            Register(fake_database).on_post(fake_request, fake_response)
+            self.logger.info("STEP: Verify that a 400 Bad Request is returned.")
+            self.assertEqual(
+                fake_response.fake_responses.get("status"), falcon.HTTP_400
+            )
```
Comment (t-persson) [inline tests/test_webserver_register.py:118]: These should be moved outside of the `assertRaises` or they do not run.

**Review by andjoe-axis — APPROVED** [review]

**Review by fredjn — CHANGES_REQUESTED** [review]

File: test-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+mock
```
[the remark this answers is not in the document]
Comment (t-persson) [reply test-requirements.txt:1]: Will be needed later.

**Review by fredjn — APPROVED** [review]

**Review by nordixinfra — COMMENTED** [review]
- Build Status: **SUCCESS**
- Build URL: [Nordix Jenkins](https://jenkins.nordix.org/job/etos-environment-provider-build-merge-ubuntu2004-main/8/)
- Command to pull image:
    `docker pull registry.nordix.org/eiffel/etos-environment-provider:dev`
    or
    `podman pull registry.nordix.org/eiffel/etos-environment-provider:dev`

**Review by nordixinfra — COMMENTED** [review]
- Build Status: **SUCCESS**
- Build URL: [Nordix Jenkins](https://jenkins.nordix.org/job/etos-environment-provider-build-merge-ubuntu2004-main/9/)
- Command to pull image:
    `docker pull registry.nordix.org/eiffel/etos-environment-provider:dev`
    or
    `podman pull registry.nordix.org/eiffel/etos-environment-provider:dev`

**Pull request merged**